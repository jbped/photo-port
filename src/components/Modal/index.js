import React from 'react'

function Modal({ currentPhoto, onClose }) {
    const {name, description, category, index} = currentPhoto;
    return (
        <div className="modalBackdrop" onClick={onClose}>
            <div className="modalContainer">
                <h3 className="modalTitle">{name}</h3>
                <img src={require(`../../assets/small/${category}/${index}.jpg`).default} alt={category} />
                <p>{description}</p>
                <button type="button" onClick={onClose}>Close this Modal</button>
            </div>
        </div>
    );
}

export default Modal;