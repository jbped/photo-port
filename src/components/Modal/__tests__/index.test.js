import React from 'react';
import { cleanup, render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Modal from '..';

const mockToggleModal = jest.fn();

const currentPhoto = {
    name: "Astartes",
    category: "commercial",
    description: "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Ex, tenetur soluta ut voluptates eligendi molestiae placeat sint possimus illum ullam laboriosam atque praesentium ducimus quas expedita commodi, est ipsum optio.",
    index: 1
}

afterEach(cleanup);

describe('Modal Component', () => {
    it('renders', () => {
        render(<Modal currentPhoto={currentPhoto} />)
    });

    it('matches snapshot of DOM node structure', () => {
        const {asFragment} = render(<Modal currentPhoto={currentPhoto} />);
        expect(asFragment()).toMatchSnapshot()
    }); 
});

describe('click event', () => {
    it('calls onClose handler', () => {
        const { getByText } = render(<Modal currentPhoto={currentPhoto} onClose={mockToggleModal} />)
        fireEvent.click(getByText("Close this Modal"))
        expect(mockToggleModal).toHaveBeenCalled();
    })
})
