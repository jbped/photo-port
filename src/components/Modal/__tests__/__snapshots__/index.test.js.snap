// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component matches snapshot of DOM node structure 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
      >
        Astartes
      </h3>
      <img
        alt="commercial"
      />
      <p>
        Lorem ipsum dolor sit amet consectetur, adipisicing elit. Ex, tenetur soluta ut voluptates eligendi molestiae placeat sint possimus illum ullam laboriosam atque praesentium ducimus quas expedita commodi, est ipsum optio.
      </p>
      <button
        type="button"
      >
        Close this Modal
      </button>
    </div>
  </div>
</DocumentFragment>
`;
